src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PreemptiveCacheTest.kt

```diff
@@ -214,6 +215,49 @@ class PreemptiveCacheTest : IntegrationTestBase() {
     assertThat(secondResult is ClientResult.Success).isTrue
     assertThat((secondResult as ClientResult.Success).body).isEqualTo(offenderDetailsResponse)
   }
+
+  @Test
+  fun `A cache failure due to missing data is handled correctly`() {
+    val firstCallInstant = Instant.parse("2023-04-25T11:35:00+01:00")
+    val fourSecondsLaterInstant = Instant.parse("2023-04-25T11:35:04+01:00")
+
+    val crn = "ABCD1234"
+
+    val offenderDetailsResponse = OffenderDetailsSummaryFactory()
+      .withCrn(crn)
+      .produce()
+
+    mockSuccessfulGetCallWithJsonResponse(
+      url = "/secure/offenders/crn/$crn",
+      responseBody = offenderDetailsResponse,
+    )
+
+    every { Instant.now() } returns firstCallInstant
+
+    // The first call should make an upstream request
+    val firstResult = preemptivelyCachedClient.getOffenderDetailSummaryWithCall(crn)
+    assertThat(firstResult is ClientResult.Success).isTrue
+    assertThat((firstResult as ClientResult.Success).body).isEqualTo(offenderDetailsResponse)
+    wiremockServer.verify(exactly(1), getRequestedFor(urlEqualTo("/secure/offenders/crn/${offenderDetailsResponse.otherIds.crn}")))
+
+    // Subsequent calls up to successSoftTtlSeconds should return the cached value without making an upstream request
+    every { Instant.now() } returns fourSecondsLaterInstant
+
+    val secondResult = preemptivelyCachedClient.getOffenderDetailSummaryWithCall(crn)
+    assertThat(secondResult is ClientResult.Success).isTrue
+    assertThat((secondResult as ClientResult.Success).body).isEqualTo(offenderDetailsResponse)
+    wiremockServer.verify(exactly(1), getRequestedFor(urlEqualTo("/secure/offenders/crn/${offenderDetailsResponse.otherIds.crn}")))
```

Comment:
I was expecting to read that we'd expect no external call to be made, since we expect the cache to be the one to respond. Should this be exactly(0)?

Reply:
It's `exactly(1)` because of the first call, which does make the upstream request - Wiremock keeps track of the total number of times that the endpoint was requested.